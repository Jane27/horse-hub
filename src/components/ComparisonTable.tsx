import React from "react";
import styled from "styled-components";
import get from "lodash/get";

const Row = styled.div`
  display: flex;
  flex-direction: row;
  width: 100%;
`;

const Cell = styled.div`
  display: flex;
  height: 35px;
  background-color: white;
  color: black;
  font-size: 14px;
  border: 1px solid black;
  width: 80%;
`;
export const Text = styled.p`
  margin: auto;
  text-align: center;
`;

Text.displayName = "value";


export type TComparison = {
  fields: { label: string; path: string }[];
  items: object[];
};

const ComparisonTable: React.FC<TComparison> = ({ fields, items }) => {

  console.log(fields)
  return (
    <div>
      {fields.map(({ label, path }, idx) => {
        const values = items.map((item, idx) => (
          <Cell key={idx}>
            <Text>{get(item, path, "")}</Text>
          </Cell>
        ));
        return (
          <Row key={idx}>
            <Cell>
              <Text>{label}</Text>
            </Cell>
            {values}
          </Row>
        );
      })}
    </div>
  );
};

export default ComparisonTable;
