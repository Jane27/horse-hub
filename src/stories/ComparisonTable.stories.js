import React from 'react';
import { withKnobs, object } from '@storybook/addon-knobs/react';
import { ComparisonTable } from '../components';

export default {
  title: 'ComparisonTable',
  decorators: [withKnobs],
  component: ComparisonTable,
};

const fields = [
  {
    label: "Name",
    path: "name",
  },
  {
    label: "Favourite Food",
    path: "profile.favouriteFood",
  },
  {
    label: "Height",
    path: "profile.physical.height",
  },
  {
    label: "Weight",
    path: "profile.physical.weight",
  },
];

const options = [
  {
    name: "Red boy",
    profile: {
      favouriteFood: "apple",
      physical: {
        height: 300,
        weight: 60
      }
    }
  },
  {
    name: "Purple girl",
    profile: {
      favouriteFood: "orange",
      physical: {
        height: 70,
        weight: 250
      }
    }
  },
  {
    name: "Green girl",
    profile: {
      favouriteFood: "orange",
      physical: {
        height: 70,
        weight: 250
      }
    }
  }
]




export const CompareTItems = () => {
  const horse = object ("horse", options)
  const compareFields = object ("compareFields", fields)

return <ComparisonTable fields={compareFields} items={horse} />};


