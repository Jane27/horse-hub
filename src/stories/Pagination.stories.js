import React from 'react';
import { withKnobs, number } from '@storybook/addon-knobs/react';
import { Pagination } from '../components';


export default {
  title: 'Pagination',
  decorators: [withKnobs],
  component: Pagination,
};


export const simple = () =>{
  const total = number("total", 48);
  const currentPage = number("currentPage", 0);  
  const pageSize = number("pageSize", 10);
return <Pagination total={total} initialPage={currentPage} pageSize={pageSize} >Hello Button</Pagination>;

} 

