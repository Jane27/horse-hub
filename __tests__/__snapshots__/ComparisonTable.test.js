import { shallow } from 'enzyme';
import { CompareTItems } from '../../src/stories/ComparisonTable.stories'

describe('WithMenu', () => {
  it('Should 16 value in the table', () => {
    let output = shallow(CompareTItems());
    expect(output).toMatchSnapshot();
    console.log(output.debug())
    expect(output.find('value').first().text()).toEqual('Name');
    expect(output.find('value')).toHaveLength(16);
    expect(output.find('value').last().text()).toEqual('250');
  });
});

