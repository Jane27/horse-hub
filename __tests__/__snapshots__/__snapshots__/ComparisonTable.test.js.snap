// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithMenu Should have the icon title and dropdown 1`] = `
<div>
  <styled.div
    key="0"
  >
    <styled.div>
      <value>
        Name
      </value>
    </styled.div>
    <styled.div
      key="0"
    >
      <value>
        Red boy
      </value>
    </styled.div>
    <styled.div
      key="1"
    >
      <value>
        Purple girl
      </value>
    </styled.div>
    <styled.div
      key="2"
    >
      <value>
        Green girl
      </value>
    </styled.div>
  </styled.div>
  <styled.div
    key="1"
  >
    <styled.div>
      <value>
        Favourite Food
      </value>
    </styled.div>
    <styled.div
      key="0"
    >
      <value>
        apple
      </value>
    </styled.div>
    <styled.div
      key="1"
    >
      <value>
        orange
      </value>
    </styled.div>
    <styled.div
      key="2"
    >
      <value>
        orange
      </value>
    </styled.div>
  </styled.div>
  <styled.div
    key="2"
  >
    <styled.div>
      <value>
        Height
      </value>
    </styled.div>
    <styled.div
      key="0"
    >
      <value>
        300
      </value>
    </styled.div>
    <styled.div
      key="1"
    >
      <value>
        70
      </value>
    </styled.div>
    <styled.div
      key="2"
    >
      <value>
        70
      </value>
    </styled.div>
  </styled.div>
  <styled.div
    key="3"
  >
    <styled.div>
      <value>
        Weight
      </value>
    </styled.div>
    <styled.div
      key="0"
    >
      <value>
        60
      </value>
    </styled.div>
    <styled.div
      key="1"
    >
      <value>
        250
      </value>
    </styled.div>
    <styled.div
      key="2"
    >
      <value>
        250
      </value>
    </styled.div>
  </styled.div>
</div>
`;
